monai/utils/utils_random_generator_adaptor.py

```diff
@@ -0,0 +1,144 @@
+        ...
+
+    def uniform(self, low: float = 0.0, high: float = 1.0, size: ShapeLike | None = None) -> NdarrayTensor:
+        ...
+
+
+class LegacyRandomStateAdaptor(SupportsRandomGeneration):
+    random_staters: np.random.RandomState
+
+    def __init__(self, seed: int | None = None, random_state: np.random.RandomState | None = None):
+        if random_state is not None and seed is not None:
+            raise ValueError("Cannot specify both rs and seed.")
+        self.random_state = np.random.RandomState(seed=seed) if random_state is None else random_state
+
+    def integers(
+        self, low: int, high: int | None = None, size: ShapeLike | None = None, dtype=DtypeLike, endpoint: bool = False
+    ) -> NdarrayTensor:
+        if endpoint:
+            if high is not None:
+                high += 1
+            else:
+                low += 1
+        return self.random_state.randint(low=low, high=high, size=size, dtype=dtype)
+
+    def random(self, size: ShapeLike | None = None, dtype=DtypeLike, out: NdarrayTensor | None = None) -> NdarrayTensor:
+        if out is not None:
+            raise NotImplementedError("out is not implemented")
+        if dtype is not None:
+            raise NotImplementedError("dtype is not implemented")
+        return self.random_state.random(size)
+
+    def choice(
+        self,
+        a: NdarrayTensor,
+        size: ShapeLike | None = None,
+        replace: bool = True,
+        p: NdarrayTensor | None = None,
+        axis: int = 0,
+        shuffle: bool = True,
+    ) -> NdarrayTensor:
+        if axis != 0:
+            raise NotImplementedError("axis is not implemented")
+        if not shuffle:
+            raise NotImplementedError("shuffle is not implemented")
+        return self.random_state.choice(a, size, replace, p)
+
+    def permutation(self, x: NdarrayTensor, axis: int = 0) -> NdarrayTensor:
+        if axis != 0:
+            raise NotImplementedError("axis is not implemented")
+        return self.random_state.permutation(x)
+
+    def shuffle(self, x: NdarrayTensor, axis: int = 0) -> None:
+        if axis != 0:
+            raise NotImplementedError("axis is not implemented")
+        return self.random_state.shuffle(x)
+
+    def multinomial(
+        self, n: _ArrayLikeInt_co, pvals: _ArrayLikeFloat_co, size: ShapeLike | None = None
+    ) -> NdarrayTensor:
+        return self.random_state.multinomial(n, pvals, size)
+
+    def normal(self, loc: float = 0.0, scale: float = 1.0, size: ShapeLike | None = None) -> NdarrayTensor:
+        return self.random_state.normal(loc, scale, size)
+
+    def uniform(self, low: float = 0.0, high: float = 1.0, size: ShapeLike | None = None) -> NdarrayTensor:
+        return self.random_state.uniform(low, high, size)
+
+
+def handle_legacy_random_state(
+    rand_state: np.random.RandomState | None = None,
+    generator: SupportsRandomGeneration | None = None,
+    return_legacy_default_random: bool = False,
```

Comment:
Can the user directly change to the Generator by setting the flag?

Reply:
This function is more of a internal utility function (should probably be prefixed with _) during the migration to be used in the utility functions that accept a random_state (e.g. get_random_patch). The return_legacy_default_random is to return the default value as it is currently done (If there ends up being no use case where the default is not required, the parameter could be removed as well).

To change the default generator for transforms the place would be to change the [_default_random_generator_factory](https://github.com/john-zielke-snkeos/MONAI/blob/c564deda2c33e487d92ef0f513e89f7133b1cafe/monai/transforms/transform.py#L175) variable to numpy.random.default_rng